pkg/controller/secretgenerator/secretgenerator_controller.go

```diff
@@ -130,6 +134,18 @@ func (r *ReconcileSecretGenerator) Reconcile(request reconcile.Request) (reconci
 			}
 
 			err = r.client.Create(context.TODO(), generatedSecret)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if _, exists := secretAnnotations[SECRET_OAUTH_CLIENT_ANNOTATION]; exists {
```

Comment:
You can simplify this by moving this logic into the `Secret` object:

```golang
type Secret struct {
	Name                    string
	Type                      string
	Complexity            int
	Value                     string
       OAuthClientRoute string
}

func newSecret(...) {
... // After line 85
	// Get OAuthClient route name from annotation
	if secretOAuthClientRoute, found := annotations[SECRET_OAUTH_CLIENT_ANNOTATION]; found {
		secret.OAuthClientRoute = secretOAuthClientRoute
	}
}
```

And then, in the reconciler code you can just do:

```golang
if secret.OAuthClientRoute != "" {
...
}
```